### PR TITLE
Update hamcrest

### PR DESCRIPTION
**Pull request by NotMyFault**
_🤖 This is an automatic PR for replacing the deprecated 'hamcrest-*' modules with 'hamcrest'._

  All deprecated modules' code has been merged into 'hamcrest'.
  Consumers of `hamcrest-*` should replace the dependency with `hamcrest`, which is the drop-in module containing the code of the deprecated `hamcrest-*` modules.

  Ping `@NotMyFault` if you have questions.

  _[script source](https://github.com/NotMyFault/jenkins-version-updater)_

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -5,7 +5,7 @@
   <parent>
     <groupId>org.jenkins-ci.plugins</groupId>
     <artifactId>plugin</artifactId>
-    <version>4.49</version>
+    <version>4.64</version>
```
Comment (NotMyFault) [inline pom.xml:8]:
```suggestion
    <version>4.49</version>
```

File: pom.xml [hunk 2]
```diff
@@ -121,7 +121,7 @@
     </dependency>
     <dependency>
       <groupId>org.hamcrest</groupId>
-      <artifactId>hamcrest-core</artifactId>
+      <artifactId>hamcrest</artifactId>
```
Comment (NotMyFault) [inline pom.xml:124]:
```suggestion
      <artifactId>hamcrest</artifactId>
      <version>2.2</version>
```
To be removed when updating the parent pom.

Comment (lifeofguenter) [reply pom.xml:124]: apologies, does it mean we can get rid of it completely?

Comment (NotMyFault) [reply pom.xml:124]: > apologies, does it mean we can get rid of it completely?

No, just the `<version>2.2</version>` part. Plugin pom versions newer than 4.64 contain `org.hamcrest:hamcrest:2.2`, so this override is no longer needed.

**Comment by lifeofguenter** [comment]
closing this then - thanks, fortunately got some help to push many dependencies.

**Comment by NotMyFault** [comment]
Oic, was already done in https://github.com/jenkinsci/bitbucket-branch-source-plugin/commit/50d0c8dba3a6b3c94589a5d8abae3b2ef792653b 👍

**Pull request closed without merging**